fabulaws/library/wsgiautoscale/api.py

```diff
@@ -466,7 +466,12 @@ def _new(deployment, environment, role, avail_zone=None, count=1, **kwargs):
                      tags=tags, volume_size=vol_size, volume_type=vol_type,
                      deploy_user=env.deploy_user, security_groups=sec_grps,
                      **extra_args)
-        server.setup()
+        try:
+            server.setup()
+        except:
+            if terminate_on_failure:
+                server.terminate()
```

Comment:
Oh, I guess `server.terminate()` will eventually return and then the exception will be raised - though it might be confusing if it's logged long after it actually happened.  Maybe logging it first would have some advantages.